### PR TITLE
POPJAM-3015: Add NavBarTitle skin

### PR DESCRIPTION
**Pull request by Tbruni85**
## JIRA
https://jira.superawesome.tv/browse/POPJAM-3015

## Description
Skin option added to *NavBarTitle*

### Default Skin
simulator screen shot - iphone x - 2018-05-25 at 16 57 45

### Custom Skin
simulator screen shot - iphone x - 2018-05-25 at 16 57 47

### CODE REVIEW

File: PopJamUI/Classes/GenericViews/LiveHeader.swift [hunk 1]
```diff
@@ -63,15 +74,14 @@ public class LiveHeader: UIView {
         // icon
         icon = UIImageView(frame: .zero)
         icon.accessibilityIdentifier = "icon_image_view"
+        icon.image = UIImage.init(named: theme.iconName!)
```
Comment (me2k) [inline PopJamUI/Classes/GenericViews/LiveHeader.swift]: Explicit unwrapping.... Can't you check the name exists then pass it in?

File: PopJamUI/Classes/GenericViews/LiveHeader.swift [hunk 2]
```diff
@@ -63,15 +74,14 @@ public class LiveHeader: UIView {
...
         container.addSubview(icon)
         
         // label
         titleLabel = UILabel(frame: .zero)
         titleLabel.accessibilityIdentifier = "title_label_view"
-        titleLabel.textColor = Color.textColour1.value
-        titleLabel.font = UIFont(name: "TitilliumWeb-Bold", size: 16)
-        titleLabel.textAlignment = .center
-        titleLabel.text = title
+        titleLabel.applySkin(skin: theme.title!)
```
Comment (me2k) [inline PopJamUI/Classes/GenericViews/LiveHeader.swift]: Same again here, check the object exists before explicitly unwrapping

File: PopJamUI/Classes/Skins/FontMap.swift [hunk 3]
```diff
@@ -59,6 +60,8 @@ extension Font {
             instanceFont = UIFont().Helvetica8()
         case .H7:
             instanceFont = UIFont().TitilliumWeb_Bold12()
+        case .H8:
```
Comment (me2k) [inline PopJamUI/Classes/Skins/FontMap.swift]: Lower case here for the enum cases

File: PopJamUI/Classes/Utilis/Fonts.swift [hunk 4]
```diff
@@ -163,6 +163,15 @@ extension UIFont {
         }
     }
     
+    // TitilliumWeb-Bold 16
+    func TitilliumWeb_Bold16() -> UIFont {
+        if Locale.current.languageCode == "vn" {
+            return UIFont.init(name: "Itoya-Bold", size: 8)!
```
Comment (me2k) [inline PopJamUI/Classes/Utilis/Fonts.swift]: No explicit unwrapping here, use this: `return UIFont.init(name: "Itoya-Bold", size: 8) ?? UIFont.systemFont(ofSize: 8)`

File: PopJamUI/Classes/GenericViews/LiveHeader.swift (hunk 1 above)
Comment (Tbruni85) [reply PopJamUI/Classes/GenericViews/LiveHeader.swift]: I've updated the protocol, no more optionals ;)

File: PopJamUI/Classes/GenericViews/LiveHeader.swift [hunk 5]
```diff
@@ -36,16 +36,27 @@ public class LiveHeader: UIView {
         setUpView()
     }
     
+    public init(frame: CGRect, theme: UINavigationBarTitleSkinProtocol?) {
+        if let viewTheme = theme {
```
Comment (me2k) [inline PopJamUI/Classes/GenericViews/LiveHeader.swift]: change to:
`if let viewTheme = theme as? UINavigationBarTitleSkinModel {`

**Pull request merged**